api/userController.go

```diff
@@ -0,0 +1,92 @@
+package api
+
+import (
+	"host/types"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
```

Comment:
I'd really love it if you use gorilla. it has a similar API to the standard library and I dont think we need gin for performance here